django/db/models/indexes.py

```diff
@@ -4,12 +4,11 @@
 
 __all__ = ['Index']
 
-# The max length of the names of the indexes (restricted to 30 due to Oracle)
-MAX_NAME_LENGTH = 30
-
 
 class Index:
     suffix = 'idx'
+    # Default max length of the names of the indexes (restricted to 30 due to Oracle)
+    max_name_length = 30
```

Comment:
Maybe a new `Index` property (without special treatment for the `BrinIndex`) will be a more universal solution, i.e. sth like:
```python
@property
def max_name_length(self):
    return connection.ops.max_name_length() or 30
```